### PR TITLE
NJ/update: Added new button type

### PR DESCRIPTION
**Pull request by njazuli-deriv**
Changes:

-  New button type for Hero
-  Cta fixes for Hero - Deriv Academy

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [X] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/academy/index.js [hunk 1]
```diff
@@ -381,19 +378,15 @@ const DerivBlog = ({ data }) => {
...
-                                rel="noopener noreferrer"
-                                target="_blank"
-                            >
+                            <Flex key={page_data.id}>
                                 <Hero
                                     imageData={page_data.desktop_hero.imageFile}
                                     imageAlt={page_data?.desktop_hero?.description}
                                     title={page_data.heading}
                                     description={page_data.sub_heading}
+                                    href={page_data.link}
```
Comment (sean-binary) [inline src/pages/academy/index.js:387]: We also need to pass the CTA button text coming from the API. We'll need to create the field for the CMS.

File: src/pages/academy/components/_hero.js [hunk 2]
```diff
@@ -48,18 +49,22 @@ const Wrapper = styled(Flex)`
...
 
-const Hero = ({ imageAlt, imageData, title, description }) => {
+const Hero = ({ cta_text, href, imageAlt, imageData, title, description }) => {
     const [is_mobile] = useBrowserResize()
     const linear_bg_value = is_mobile
         ? `linear-gradient(76.78deg, #000000 30.72%, rgba(0, 0, 0, 0) 97.58%)`
         : `linear-gradient(66.11deg, #000000 24.94%, rgba(0, 0, 0, 0) 83.1%)`
 
     const backgroundFluidImageStack = [imageData.childImageSharp.fluid, linear_bg_value].reverse()
+    const cta_link = cta_text ? cta_text : 'Learn more 1'
```
Comment (suthesan-deriv) [inline src/pages/academy/components/_hero.js:67]: Is it expected to be "Learn more 1"? 👀

File: src/components/form/button.js [hunk 3]
```diff
@@ -82,6 +83,26 @@ export const SharedButtonStyle = css`
                     color: var(--color-red);
                 }
             `
+        if (props.hero)
+            return css`
+                border: 2px solid var(--color-red);
```
Comment (sean-binary) [inline src/components/form/button.js:88]: `Border` is accounted for the height of the element for the top and bottom as well, so if we want a 64px button, you need to either use `box-shadow` instead or minus 2px from the vertical padding, then we should be able to remove the height prop.

File: src/pages/academy/components/_hero.js (hunk 2 above)
Comment (njazuli-deriv) [reply src/pages/academy/components/_hero.js]: my bad..fixed in the latest commit

File: src/components/form/button.js (hunk 3 above)
Comment (njazuli-deriv) [reply src/components/form/button.js]: I've removed the height props and added the `box-shadow` to replace border.

File: src/pages/academy/components/_hero.js [hunk 4]
```diff
@@ -71,27 +76,34 @@ const Hero = ({ imageAlt, imageData, title, description }) => {
...
                         font_size={['20px', '14px']}
                         line_height={['30px', '24px']}
-                        max_width={['684px', '328px']}
+                        max_width={['636px', '328px']}
                     >
                         {description}
                     </TextWrapper>
+                    <Flex jc="flex-start" height="unset" mt="24px" tabletL={{ mt: '16px' }}>
+                        <LinkButton id="hero-article" to={href} width="auto" hero>
+                            {cta_link}
```
Comment (sean-binary) [inline src/pages/academy/components/_hero.js:93]:
```suggestion
                            {button_text}
```
this would be more descriptive.

**Review by sean-binary — APPROVED** [review]

**Pull request merged**